### PR TITLE
Transition kubernetes client 7.0 module to instrumenter API

### PR DESCRIPTION
**Pull request by jack-berg**
Contributing to Issue #2713.

### CODE REVIEW

File: instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/CurrentState.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  public static void set(Context parentContext, Context context, Scope scope, Request request) {
+    CURRENT.set(new CurrentState(parentContext, context, scope, request));
+  }
+
+  @Nullable
+  public static CurrentState remove() {
+    CurrentState contextAndScope = CURRENT.get();
+    CURRENT.remove();
+    return contextAndScope;
```
Comment (trask) [inline instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/CurrentState.java:40]:
```suggestion
    CurrentState currentState = CURRENT.get();
    CURRENT.remove();
    return currentState;
```

File: instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesClientSingletons.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    INSTRUMENTER =
+        Instrumenter.<Request, ApiResponse<?>>newBuilder(
+                GlobalOpenTelemetry.get(),
+                "io.opentelemetry.kubernetes-client-7.0",
+                spanNameExtractor)
+            .setSpanStatusExtractor(HttpSpanStatusExtractor.create(httpAttributesExtractor))
+            .addAttributesExtractor(httpAttributesExtractor)
+            .addAttributesExtractor(new KubernetesNetAttributesExtractor())
+            .addAttributesExtractor(new KubernetesExperimentalAttributesExtractor())
+            .newInstrumenter(alwaysClient());
```
Comment (trask) [inline instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesClientSingletons.java:44]: can you add a comment here briefly explaining why `newClientInstrumenter` doesn't work?

**Review by trask — APPROVED** [review]
thx 👍

File: instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesExperimentalAttributesExtractor.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import okhttp3.Request;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class KubernetesExperimentalAttributesExtractor
+    extends AttributesExtractor<Request, ApiResponse<?>> {
+  @Override
+  protected void onStart(AttributesBuilder attributes, Request request) {
+    if (!captureExperimentalSpanAttributes()) {
```
Comment (mateuszrzeszutek) [inline instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesExperimentalAttributesExtractor.java:20]: If experimental attributes are turned off you could just not add this `AttributesExtractor` to the instrumenter.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**